File: src/server/pds.js
const { spawn } = require('child_process');
const path = require('path');

const pythonPath = path.join(__dirname, 'pds.py');
const py = spawn('python', [pythonPath]);

py.stdout.on('data', (data) => {
    console.log('Python dice:', data.toString());
});

py.stderr.on('data', (data) => {
    console.error('Error de Python:', data.toString());
});

function enviarDatos(datos) {
    py.stdin.write(JSON.stringify(datos) + '\n');
}

module.exports = {
    enviarDatos
};

const datosParaPython = {
    tau_x: getVal(10),
    tau_y: getVal(11),
    tau_z: getVal(12),
    KalmanAngleRoll: getVal(13),
    KalmanAnglePitch: getVal(14),
};

pds.enviarDatos(datosParaPython);

File: src/server/pds.py
import sys
import json

for line in sys.stdin:
    try:
        data = json.loads(line)
        print(
            f"Recibido de Node.js: tau_x={data['tau_x']}, tau_y={data['tau_y']}, tau_z={data['tau_z']}, "
            f"KalmanAngleRoll={data['KalmanAngleRoll']}, KalmanAnglePitch={data['KalmanAnglePitch']}"
        )
        sys.stdout.flush()
    except Exception as e:
        print(f"Error al procesar datos: {e}")
        sys.stdout.flush()